package com.groupsavings.exception;

import java.time.LocalDateTime;

public class ErrorResponse {

	private int status;

	private String message;

	private String path;

	private LocalDateTime timestamp;

	private String errorCode;

	public ErrorResponse(int status, String message, String path) {
		this.status = status;
		this.message = message;
		this.path = path;
		this.timestamp = LocalDateTime.now();
	}

	public int getStatus() {
		return status;
	}

	public void setStatus(int status) {
		this.status = status;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}

	public String getPath() {
		return path;
	}

	public void setPath(String path) {
		this.path = path;
	}

	public LocalDateTime getTimestamp() {
		return timestamp;
	}

	public void setTimestamp(LocalDateTime timestamp) {
		this.timestamp = timestamp;
	}

	public String getErrorCode() {
		return errorCode;
	}

	public void setErrorCode(String errorCode) {
		this.errorCode = errorCode;
	}

}
